pending-registrations/src/main/java/engineering/everest/lhotse/registrations/domain/OrganizationRegistrationSaga.java

```diff
@@ -83,37 +84,37 @@ public void on(OrganizationRegistrationReceivedEvent event) {
                 event.getOrganizationId(), event.getRegisteringContactEmail(), event.getOrganizationName()));
     }
 
-    @SagaEventHandler(associationProperty = ASSOCIATION_PROPERTY)
+    @SagaEventHandler(associationProperty = CONFIRMATION_CODE)
     public void on(OrganizationRegistrationConfirmedEvent event) {
         // The business may require that each email address be only available for registration once. In this case,
         // the target aggregate identifier may be the user email address (possibly with an additional prefix).
         if (usersReadService.hasUserWithEmail(registeringUserEmail)) {
             commandGateway.send(new CancelConfirmedRegistrationUserEmailAlreadyInUseCommand(
                     event.getRegistrationConfirmationCode(), organizationId, registeringUserId, registeringUserEmail));
         } else {
-            commandGateway.send(new CreateRegisteredOrganizationCommand(organizationId, registeringUserId, organizationName,
-                    street, city, state, country, postalCode, websiteUrl, registeringUserDisplayName, phoneNumber, registeringUserEmail));
-
-            commandGateway.send(new CreateUserForNewlyRegisteredOrganizationCommand(registeringUserId, organizationId,
+            var registeredOrganizationCommand = new CreateRegisteredOrganizationCommand(organizationId, registeringUserId, organizationName,
+                    street, city, state, country, postalCode, websiteUrl, registeringUserDisplayName, phoneNumber, registeringUserEmail);
+            var createUserCommand = new CreateUserForNewlyRegisteredOrganizationCommand(registeringUserId, organizationId,
                     event.getRegistrationConfirmationCode(), registeringUserEmail, registeringUserEncodedPassword,
-                    registeringUserDisplayName));
+                    registeringUserDisplayName);
+            var promoteUserToOrganizationAdminCommand = new PromoteUserToOrganizationAdminCommand(organizationId, registeringUserId);
+            var completeRegistrationCommand = new CompleteOrganizationRegistrationCommand(event.getRegistrationConfirmationCode(),
+                    organizationId, registeringUserId);
+
+            commandGateway.send(registeredOrganizationCommand)
+                    .thenApply(fun -> commandGateway.sendAndWait(createUserCommand))
+                    .thenApply(fun -> commandGateway.sendAndWait(promoteUserToOrganizationAdminCommand))
+                    .thenApply(fun -> commandGateway.send(completeRegistrationCommand));
```

Comment:
The first command needs to be `sendAndWait` as well. Otherwise the whole event handling is still async. If there is any extra delay in Organization or User aggregate/handler (e.g. can be simulated adding `Thread.sleep(1000)` in `UsersEventHandler#on(UserPromotedToOrganizationAdminEvent event)`)  , the test would still fail.

I suggest change it to the followings:
```java
commandGateway.sendAndWait(registeredOrganizationCommand);
commandGateway.sendAndWait(createUserCommand);
commandGateway.sendAndWait(promoteUserToOrganizationAdminCommand);
commandGateway.send(completeRegistrationCommand);
```
Note the last command needs to be async, or we would run into deadlock.